### PR TITLE
Problem: not updated to defi-wallet-core v0.3.1 (fix #297)

### LINKED ISSUE #297: Problem: not updated to defi-wallet-core v0.3.1
Solution: update to latest defi-wallet-core

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: update to latest

### CODE REVIEW

**Comment by leejw51crypto** [comment]
working on windows build issue

File: .github/workflows/cpp-lint.yml [hunk 1]
```diff
@@ -9,6 +9,10 @@ on:
 jobs:
   cpp-linter:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        compiler: [clang++-14]
```
Comment (tomtau) [inline .github/workflows/cpp-lint.yml]: it seems this matrix/environment isn't being used? the compiler is always hardcoded to 14 elsewhere?

**Comment by elecming** [comment]
Seems we are fixing the same issue: https://github.com/cronos-labs/play-cpp-sdk/pull/292?

**Comment by elecming** [comment]
But as I mentioned on https://github.com/cronos-labs/play-cpp-sdk/pull/292, `.github/workflows/cpp-lint.yml` should be removed, because the linter checking should be conducted after build.

**Comment by elecming** [comment]
You can check the failure details inside https://github.com/cronos-labs/play-cpp-sdk/actions/runs/3635125599/jobs/6133931849

**Comment by elecming** [comment]
I deleted cpp-linter.yml on https://github.com/cronos-labs/play-cpp-sdk/pull/292 to avoid linter issues (it is incorrect setup as I mentioned), please help review and merge before fixing the linter issue. So that we would be break the build because the linter issue.

**Review by elecming — CHANGES_REQUESTED** [review]
Please check https://github.com/cronos-labs/play-cpp-sdk/pull/292 and merge it first.

**Comment by leejw51crypto** [comment]
ok

**Review by elecming — DISMISSED** [review]

File: .github/workflows/cpp-lint.yml (hunk 1 above)
Comment (leejw51crypto) [reply .github/workflows/cpp-lint.yml]: yes, i'll remove it.

**Comment by tomtau** [comment]
maybe tag defi-wallet-core v0.3.1 first?

**Comment by leejw51crypto** [comment]
yes, i'm checking now

**Comment by leejw51crypto** [comment]
rebasing now

**Comment by leejw51crypto** [comment]
rebased
using defi-wallet-core 0.3.1

**Comment by elecming** [comment]
Please update https://github.com/cronos-labs/play-cpp-sdk/blob/main/play-cpp-sdk/Cargo.toml#L10 as well.

**Comment by leejw51crypto** [comment]
ok

**Review by tomtau — APPROVED** [review]

File: play-cpp-sdk/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "play-cpp-sdk"
-version = "0.0.11-alpha"
+version = "0.0.12-alpha"
```
Comment (elecming) [inline play-cpp-sdk/Cargo.toml:3]: Still missing
`defi-wallet-core-cpp = { path = "../defi-wallet-core-rs/bindings/cpp/", version = "0.3.1"}`
or
`defi-wallet-core-cpp = { path = "../defi-wallet-core-rs/bindings/cpp/", version = "0.3"}`
?

**Review by elecming — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 ## [Unreleased]
+
+## [v0.0.12-alpha] - 2022-12-7
+- Dynamic Contract C++ Bindings : Encode,Call,Send
+- Minting C++ Example : Encode,Send
```
Comment (elecming) [inline CHANGELOG.md:6]: Changes on play-cpp-sdk itself are needed to add:
- Replace custom cxx-build with official cxx-build
- Add walletconnect registry API support
- Add app token for sepgrep CI
- Fix Walletconnect request deadlock issue
- Upgrade ethers to 1.0

**Review by elecming — CHANGES_REQUESTED** [review]

File: play-cpp-sdk/Cargo.toml (hunk 2 above)
Comment (leejw51crypto) [reply play-cpp-sdk/Cargo.toml:3]: ok

**Comment by leejw51crypto** [comment]
update changelog

**Review by elecming — APPROVED** [review]

**Pull request merged**